src/event_loop.rs

```diff
@@ -305,12 +308,17 @@ impl<T> EventLoop<T> {
     /// [^1]: `EventLoopExtWebSys::spawn()` is only available on WASM.
     #[inline]
     #[cfg(not(all(wasm_platform, target_feature = "exception-handling")))]
-    pub fn run<F>(self, mut event_handler: F) -> Result<(), EventLoopError>
+    pub fn run<F>(self, handler: F) -> Result<(), EventLoopError>
```

Comment:
I believe we could do the following:
```rust
impl<T: FnMut(Event<T>, ActiveEventLoop<'_>)> ApplicationHandler for T { ... }
```
Then we don't need to introduce an additional `EventLoop::run_with()` function.

Reply:
I think I tried that, but it wasn't backwards compatible since type inference no longer kicks in, you'd have to specify the type of the closure arguments.

E.g. `event_loop.run(|event, active| { ... })` would have to be `event_loop.run(|event: Event, active: ActiveEventLoop<'_>| { ... })`.